src/Microsoft.IdentityModel.Tokens/Validation/Validators.Audience.cs

```diff
@@ -75,92 +69,45 @@ internal static AudienceValidationResult ValidateAudience(IEnumerable<string> au
                             LogMessages.IDX10214,
-                            LogHelper.MarkAsNonPII(Utility.SerializeAsSingleCommaDelimitedString(audiencesAsList)),
-                            LogHelper.MarkAsNonPII(validationParameters.ValidAudience ?? "null"),
+                            LogHelper.MarkAsNonPII(Utility.SerializeAsSingleCommaDelimitedString(audiences)),
                             LogHelper.MarkAsNonPII(Utility.SerializeAsSingleCommaDelimitedString(validationParameters.ValidAudiences))),
                         typeof(SecurityTokenInvalidAudienceException),
                         new StackFrame(true)));
         }
 
-        private static bool AudienceIsValid(List<string> audiences, TokenValidationParameters validationParameters)
-        {
-            return AudienceIsValidReturning(audiences, validationParameters) != null;
-        }
-
-        private static string? AudienceIsValidReturning(List<string> audiences, TokenValidationParameters validationParameters)
+        private static string? AudienceIsValidReturning(List<string> audiences, ValidationParameters validationParameters)
         {
             string? validAudience = null;
-            if (!string.IsNullOrWhiteSpace(validationParameters.ValidAudience))
-                validAudience = AudiencesMatchSingle(audiences, validationParameters.ValidAudience, validationParameters.IgnoreTrailingSlashWhenValidatingAudience);
-
-            if (validAudience == null && validationParameters.ValidAudiences != null)
-            {
-                if (validationParameters.ValidAudiences is not List<string> validAudiences)
-                    validAudiences = validationParameters.ValidAudiences.ToList();
 
-                validAudience = AudiencesMatchList(audiences, validAudiences, validationParameters.IgnoreTrailingSlashWhenValidatingAudience);
-            }
+            List<string> validAudiences = validationParameters.ValidAudiences.ToList();
+            validAudience = AudiencesMatchList(audiences, validAudiences, validationParameters.IgnoreTrailingSlashWhenValidatingAudience);
 
             return validAudience;
         }
 
-        private static string? AudiencesMatchSingle(List<string> audiences, string validAudience, bool ignoreTrailingSlashWhenValidatingAudience)
-        {
-            for (int i = 0; i < audiences.Count; i++)
-            {
-                string tokenAudience = audiences[i];
-                if (string.IsNullOrWhiteSpace(tokenAudience))
-                    continue;
-
-                if (AudiencesMatch(ignoreTrailingSlashWhenValidatingAudience, tokenAudience, validAudience))
-                {
-                    if (LogHelper.IsEnabled(EventLogLevel.Informational))
-                        LogHelper.LogInformation(LogMessages.IDX10234, LogHelper.MarkAsNonPII(tokenAudience));
-
-                    return tokenAudience;
-                }
-            }
-
-            return null;
-        }
-
```

Comment:
Below, use 
```cs
for(int j = 0; j < validAudiences.Count; j++)
{
    // comparison logic
}
```

Reply:
On line 116, just check for IsNullOrEmpty.

Reply:
Updated, Thanks!